templates/mongodb.conf.erb

```diff
@@ -1,162 +1,202 @@
-<% elsif @profile == "2" -%>
-operationProfiling.mode: all
-<% end -%>
-<% if @slowms -%>
-operationProfiling.slowOpThresholdMs: <%= @slowms %>
-<% end -%>
-
-<% if @set_parameter -%>
-setParameter:
-<% [@set_parameter].flatten.each do |v| -%>
-  <%= v %>
-<% end -%>
-<% end -%>
-
-<% if @config_data -%>
-<% @config_data.each do |k,v| -%>
-<%= k %>: <%= v %>
-<% end -%>
-<% end -%>
+    <%- config_hash['processManagement']['pidFilePath'] = @pidfilepath -%>
+  <%- end -%>
+<%- end -%>
+<%- if !config_hash['storage'] -%>
+  <%- config_hash['storage'] = {} -%>
+<%- end -%>
+<%- config_hash['storage']['dbPath'] = @dbpath -%>
+<%- if !@journal.nil? -%>
+  <%- if !config_hash['storage']['journal'] -%>
+    <%- config_hash['storage']['journal'] = {} -%>
+  <%- end -%>
+  <%- config_hash['storage']['journal']['enabled'] = @journal -%>
+<%- end -%>
+<%- if @noprealloc -%>
+  <%- config_hash['storage']['preallocDataFiles'] = !@noprealloc -%>
+<%- end -%>
+<%- if @nssize -%>
+  <%- config_hash['storage']['nsSize'] = @nssize -%>
+<%- end -%>
+<%- if @directoryperdb -%>
+  <%- config_hash['storage']['directoryPerDB'] = @directoryperdb -%>
+<%- end -%>
+<%- if @smallfiles -%>
+  <%- config_hash['storage']['smallFiles'] = @smallfiles -%>
+<%- end -%>
+<%- if !@quota.nil? -%>
+  <%- if !config_hash['storage']['quota'] -%>
+    <%- config_hash['storage']['quota'] = {} -%>
+  <%- end -%>
+  <%- config_hash['storage']['quota']['enforced'] = @quota -%>
+  <%- if @quotafiles -%>
+    <%- config_hash['storage']['quota']['maxFilesPerDB'] = @quotafiles -%>
+  <%- end -%>
+<%- end -%>
+<%- if @storage_engine_internal -%>
+  <%- config_hash['storage']['engine'] = @storage_engine_internal -%>
+<%- end -%>
+<%- if !config_hash['security'] -%>
+  <%- config_hash['security'] = {} -%>
+<%- end -%>
+<%- if @auth -%>
+  <%- config_hash['security']['authorization'] = 'enabled' -%>
+<%- else -%>
+  <%- config_hash['security']['authorization'] = 'disabled' -%>
+<%- end -%>
+<%- if @keyfile -%>
+  <%- config_hash['security']['keyFile'] = @keyfile -%>
+<%- end -%>
+<%- if @noscripting -%>
+  <%- config_hash['security']['javascriptEnabled'] = @noscripting -%>
+<%- end -%>
+<%- if !config_hash['net'] -%>
+  <%- config_hash['net'] = {} -%>
+<%- end -%>
+<%- if !@ipv6.nil? -%>
+  <%- config_hash['net']['ipv6'] = @ipv6 -%>
+<%- end -%>
+<%- config_hash['net']['bindIp'] = @bind_ip.join(',') -%>
+<%- if @port -%>
+  <%- config_hash['net']['port'] = @port -%>
+<%- end -%>
+<%- if !@objcheck.nil? -%>
+  <%- config_hash['net']['wireObjectCheck'] = @objcheck -%>
+<%- end -%>
+<%- if !@rest.nil? -%>
+  <%- config_hash['net']['RESTInterfaceEnabled'] = @rest -%>
+<%- end -%>
+<%- if @maxconns -%>
+  <%- config_hash['net']['maxIncomingConnections'] = @maxconns -%>
+<%- end -%>
+<%- if !@nohttpinterface.nil? -%>
+  <%- if !config_hash['net']['http'] -%>
+    <%- config_hash['net']['http'] = {} -%>
+  <%- end -%>
+  <%- config_hash['net']['http']['enabled'] = !@nohttpinterface -%>
+<%- end -%>
```

Comment:
Probably for another PR, but I don't really like these `no`-prefixed options if the actual option is actually the reverse.

I would keep things more simple by dropping the 'no'.
Like `Optional[boolean] $enable_http_interface = undef`, but ok, that has nothing to do with this specific PR.

Reply:
Yeah I know. Just didn't want to drag that into this PR
.